### PR TITLE
chore: update babel-core to 6.0.2

### PR DESCRIPTION
**Pull request by marionebl**
This

* Updates babel-core to `v6.0.2`
* Adapts the tests to work with the new configuration
* Adds the required babel-plugin-* modules as `devDependencies`

### CODE REVIEW

**Comment by sindresorhus** [comment]
The docs also needs updating.

**Comment by marionebl** [comment]
Added information as I saw fit and removed the link to the former page about metadata in the advanced section. I was unable to find a page that replaces it. Any suggestions?

File: readme.md [hunk 1]
```diff
@@ -59,7 +61,7 @@ gulp.task('default', function () {
 ## Babel Metadata
 
 Files in the stream are annotated with a `babel` property, which
-contains the [metadata](http://babeljs.io/docs/advanced/external-helpers/#selective-builds) from `babel.transform()`.
```
Comment (sindresorhus) [inline readme.md]: We should link to the Babel docs on this one, otherwise the user has no idea what it will contain.

**Comment by sindresorhus** [comment]
The metadata example also need to use a preset. And we need to ensure `file.babel.usedHelpers` is still a thing.

**Comment by sindresorhus** [comment]
@sebmck @thejameskyle Can you take a quick look? Want to ensure we don't miss anything.

**Comment by marionebl** [comment]
As far as i can tell the notion of metadata is missing in the entire babel documentation. `metadata.usedHelpers` is still present on the results (and is mapped to `file.babel.usedHelpers`)  but http://babeljs.io/docs/usage/api/ is not exactly open about metadata sitting there.

**Comment by thejameskyle** [comment]
Looks good

**Comment by sindresorhus** [comment]
Alright. We can sort out the metadata doc thing later. Thanks @marionebl :)

**Pull request closed without merging**